hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/ByteArrayObjectDataInput.java

```diff
@@ -72,6 +73,7 @@ public void clear() {
         if (charBuffer != null && charBuffer.length > UTF_BUFFER_SIZE * 8) {
             this.charBuffer = new char[UTF_BUFFER_SIZE * 8];
         }
+        version = UNKNOWN;
```

Comment:
Should be `this.version` to be aligned with other fields (or `this.` should be cleaned up).

Reply:
good observation, done